tools/PI/DevHome.PI/Helpers/ExternalTool.cs

```diff
@@ -126,28 +132,51 @@ internal async override void InvokeTool(Window? parentWindow, int? targetProcess
             PInvoke.MessageBox(HWND.Null, errorMessage, _errorTitleText, MESSAGEBOX_STYLE.MB_ICONERROR);
         }
     }
-
-    internal async Task<Process?> InvokeToolInternal(int? pid, HWND? hwnd)
+
+    internal void ValidateOptions(ToolLaunchOptions options)
     {
-        var process = default(Process);
-
-        var parsedArguments = string.Empty;
+        // We can only get stdout if we're launching the process via CreateProcess.
+        if (options.RedirectStandardOut && ActivationType != ToolActivationType.Launch)
+        {
+            throw new InvalidOperationException("Can only redirect StandardOut with a CreateProcess launch");

```

Comment:
NIT: Change the message to "Can only redirect StandardOut with CreateProcess with a `MSIX or Protocol launch`" this way the message matches the code logic.

I can imagine a junior engineer seeing this message and not understanding that CreateProcess is possible only with protocol and MSIX launches.

Reply:
Maybe my comment looks backwards?

This checks to see if the caller requested a stdout redirect, but didn't specify a Launch (CreateProcess) activation type. If they did that, then we'd throw an exception.

You can only redirect stdout if you're doing a CreateProcess call... you can't redirect it with a protocol launch.